### PR TITLE
Add types to Stores

### PR DESCRIPTION
**Pull request by SebastienGllmt**
This is the workflow for Yoroi

UI -> Actions -> Stores -> API

People may not remember, but actually we have no types for any of the 3 arrows above connecting the 4 components (this is historically because Daedalus has no types at all and I did some work to add types but I didn't cover the whole codebase)

Since we will do a lot of refactoring of the API structure, I want types to avoid accidentally breaking everything. Unfortunately, since Flow is really bad at generics among other things, this whole thing is kind of a mess.

### CODE REVIEW

File: app/api/ada/adaWallet.js [hunk 1]
```diff
@@ -160,7 +156,7 @@ export function createAdaHardwareWallet({
 }
 
 /** Wrapper function to create new mnemonic according to bip39 */
-export const generateAdaAccountRecoveryPhrase = (): AdaWalletRecoveryPhraseResponse => (
+export const generateAdaAccountRecoveryPhrase = () => (
```
Comment (nicarq) [inline app/api/ada/adaWallet.js]:
```suggestion
export const generateAdaAccountRecoveryPhrase:  Array<string> = () => (
```

File: app/api/ada/index.js [hunk 2]
```diff
@@ -134,16 +113,130 @@ import {
...
+
+// refreshTransactions
+
+export type GetTransactionsRequestOptions = {
+  skip: number,
+  limit: number,
+};
+export type GetTransactionsRequest = {
+  walletId: string,
+  ...$Shape<GetTransactionsRequestOptions>
```
Comment (nicarq) [inline app/api/ada/index.js]: this is cool. I didn’t know `$Shape`

File: app/api/localStorage/index.js [hunk 3]
```diff
@@ -58,7 +60,7 @@ export default class LocalStorageApi {
 
   setTermsOfUseAcceptance = (): Promise<void> => new Promise((resolve, reject) => {
     try {
-      localStorage.setItem(storageKeys.TERMS_OF_USE_ACCEPTANCE, true);
+      localStorage.setItem(storageKeys.TERMS_OF_USE_ACCEPTANCE, JSON.stringify(true));
```
Comment (nicarq) [inline app/api/localStorage/index.js]: 😅

Comment (SebastienGllmt) [reply app/api/localStorage/index.js]: yeah I don't know what to do about this one. Theoretically you can only save strings to localstorage (which is why the rest of this file uses `JSON` everywhere) but also saving non-strings seems to work perfectly fine

Comment (SebastienGllmt) [reply app/api/localStorage/index.js]: image

It looks like Chrome just knows what to do with it. Even through `JSON.stringify(true)` returns "true" and not `true`, the result in `localstorage` is still just the literal `true` (like before) and not "true"

Comment (nicarq) [reply app/api/localStorage/index.js]: yeah, I double checked and it works fine and we shouldn’t have any issues with compatibility. What scared me is that in the old code, it was just sent as a boolean and left to the API to decide to convert it. I think the change that you did is the *proper* way to do it.

File: app/stores/toplevel/ProfileStore.js [hunk 4]
```diff
@@ -191,7 +222,7 @@ export default class ProfileStore extends Store {
 
   getThemeVars = ({ theme }: { theme: string }) => {
     if (theme) return require(`../../themes/prebuilt/${theme}.js`);
-    return require(`../../themes/prebuilt/${THEMES.YOROI_CLASSIC}.js`); // default
+    return require(`../../themes/prebuilt/${ProfileStore.getDefaultTheme()}.js`); // default
```
Comment (nicarq) [inline app/stores/toplevel/ProfileStore.js]: I couldn’t find this method 🤔

**Comment by nicarq** [comment]
awesome work, Seba!! 👏

**Pull request merged**